[Gui Ling Ling] iP

ling-ui opened the pull request:
### Duke

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind from having to remember things you need to do. It's,

- text-based
- easy to learn
- super fast to use

All you need to do is,

1. download it from [here](https://github.com/ling-ui/ip/releases/download/A-Jar/ip.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [x]  Managing tasks
- [x]  Managing deadlines 
- [x]  Reminders

Commit:
Add Gradle support

ling-ui committed:
Implemented increment for level 1

ling-ui committed:
Implemented increment for level 2

ling-ui committed:
Implemented increment for level 3

ling-ui committed:
Implemented increment for level 4

ling-ui committed:
Implemented increment for level 5

ling-ui committed:
Implemented increment for level 6

ling-ui committed:
Fixed output for level 4

ling-ui committed:
Fixed output for level 5

ling-ui committed:
A-TextUiTesting

ling-ui committed:
Implement level 7

ling-ui committed:
Implement level 8 (Date)

ling-ui committed:
Merge pull request #1 from ling-ui/branch-Level-7

Implement level 7

ling-ui committed:
Merge branch 'master' into branch-Level-8

ling-ui committed:
Merge pull request #2 from ling-ui/branch-Level-8

Implement level 8 (Date)

ling-ui committed:
A-More OOP

ling-ui committed:
A-Packages

ling-ui committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

ling-ui committed:
Add JUnit tests

ling-ui committed:
Edit

ling-ui committed:
Add Gradle

ling-ui committed:
Add JavaDoc for most methods

ling-ui committed:
Tweaked code to comply with coding standard

ling-ui committed:
Add level 9

ling-ui committed:
Revert "Add level 9"

This reverts commit 94575ece57be0538a292693ef2fb7894ecd513b0.

ling-ui committed:
Revert "Tweaked code to comply with coding standard"

This reverts commit 5fbb859646a2915bc41342a3940e0e52ba2b43c1.

ling-ui committed:
Tweaked code to follow code style

ling-ui committed:
Level 9

ling-ui committed:
Merge pull request #3 from ling-ui/branch-A-JavaDoc

Add java doc

ling-ui committed:
Merge branch 'master' into branch-A-CodingStandard

ling-ui committed:
Merge pull request #4 from ling-ui/branch-A-CodingStandard

Tweaked code to follow code style

ling-ui committed:
Merge branch 'master' into branch-Level-9

ling-ui committed:
Merge pull request #5 from ling-ui/branch-Level-9

Add Level 9

ling-ui committed:
Re-Commit Level 9 as tag cannot be scanned in the previous commit

ling-ui committed:
Level 10- Built GUI for Duke

JasonYapzx commented on src/main/java/cs2103t/ip/duke/Duke.java:
```diff
@@ -0,0 +1,367 @@
+
+public class Duke extends Application {
+
+    public static final String FOLDER_PATH = "../data";
+    public static final String FILE_PATH = "../data/duke.txt";
```
Would it be better to make these two strings start with the PATH prefix? This would allow one to easily identify that these are PATH variables since they do belong in the same group.

JasonYapzx commented on src/main/java/cs2103t/ip/duke/Duke.java:
```diff
@@ -0,0 +1,367 @@
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    private String getResponse(String input) throws IOException {
```
For getReponse, would it be better to abstract out the parsing logic to the parser class instead? That way your main Duke class can have a shorter amount of LoC your classes follow the SLAP principle

JasonYapzx commented on src/main/java/cs2103t/ip/duke/Storage.java:
```diff
@@ -0,0 +1,89 @@
+     * @return A list comprising the tasks entered previously to Duke by the user.
+     * @throws DukeException
+     */
+    public ArrayList<Task> loadFileData() throws DukeException {
```
Would it be appropriate to abstract the parsing in this function as well?

JasonYapzx reviewed (commented):
Looks good to me. Overall neat code and well-documented classes/functions! 👍🏻

SweetPotato0213 commented on src/main/java/cs2103t/ip/duke/Ui.java:
```diff
@@ -1,79 +1,80 @@
                 "OK, I've marked this task as not done yet: \n" +
                 list.getTasks().get(taskNum - 1).getStatusIcon() +
-                list.getTasks().get(taskNum - 1).description + "\n" + LINE);
+                list.getTasks().get(taskNum - 1).description + "\n" + LINE;
```
Maybe this `LINE` can be placed in a separate line in order to stay consistent?
```suggestion
                list.getTasks().get(taskNum - 1).description + "\n" +
                LINE;
```

SweetPotato0213 commented on src/main/java/cs2103t/ip/duke/Ui.java:
```diff
@@ -1,79 +1,80 @@
+
 public class Ui {
 
     private final String LINE = "_______________________________\n";
```
I like the way you store the underscores as a private final string!

SweetPotato0213 commented on src/main/java/cs2103t/ip/duke/Ui.java:
```diff
@@ -1,79 +1,80 @@
 
-    public void showFilteredList(Tasklist tasks, int index) {
+    public String showFilteredList(Tasklist tasks, int index) {
         String list = LINE + "\n" + "Here are the matching tasks in your list: \n";
```
Maybe can change the name `list` as it will remind people of an array?

SweetPotato0213 reviewed (commented):
LGTM! Just some cosmetic changes suggested!

ling-ui committed:
Add assertions to document assumptions

User is allowed to enter empty message with no input.

Throwing an error to disallow users to enter an empty message will make it more user-friendly since it avoids confusing the users when an empty message body is shown.

Let's
* add an assertion to ensure that the user input length is more than 0, else an error will be thrown

ling-ui committed:
Improve Code Quality

ling-ui committed:
Added statistics to Duke

ling-ui committed:
Merge pull request #7 from ling-ui/branch-A-CodeQuality

Code quality

ling-ui committed:
Merge pull request #6 from ling-ui/branch-A-Assertions

Add assertions to document assumptions

ling-ui committed:
Added User Guide

ling-ui committed:
Added Statistics function to Duke

ling-ui committed:
Merge branch 'master' of https://github.com/ling-ui/ip

ling-ui committed:
Deleted unused methods and variables
